stateupgrade/state_upgrade.go

```diff
@@ -0,0 +1,45 @@
+// (c) 2023 Ava Labs, Inc. All rights reserved.
+// See the file LICENSE for licensing terms.
+
+package stateupgrade
+
+import (
+	"math/big"
+
+	"github.com/ava-labs/subnet-evm/params"
+	"github.com/ethereum/go-ethereum/common"
+)
+
+// Configure applies the state upgrade to the state.
+func Configure(stateUpgrade *params.StateUpgrade, state StateDB) error {
+	for account, upgrade := range stateUpgrade.StateUpgradeAccounts {
+		if err := upgradeAccount(account, upgrade, state); err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+// upgradeAccount applies the state upgrade to the given account.
+func upgradeAccount(account common.Address, upgrade params.StateUpgradeAccount, state StateDB) error {
+	// Create the account if it does not exist
+	if !state.Exist(account) {
+		state.CreateAccount(account)
+	}
+
+	if upgrade.BalanceChange != nil {
+		state.AddBalance(account, (*big.Int)(upgrade.BalanceChange))
+	}
+	if len(upgrade.Code) != 0 {
+		// if the nonce is 0, set the nonce to 1 as we would when deploying a contract at
+		// the address.
+		if state.GetNonce(account) == 0 {
+			state.SetNonce(account, 1)
+		}
```

Comment:
I think we only set nonce if `EIP 158` is activated
https://github.com/ava-labs/subnet-evm/blob/master/core/vm/evm.go#L526

Not sure if we should add this here though.

Reply:
Yes we should probably preserve this

Reply:
Ya let's add this check for symmetry with `create`

Reply:
Added.